File: jvm-experiments-commons/src/main/java/kr/kth/commons/timeperiod/TimeFormatter.java
package kr.kth.commons.timeperiod;

import org.joda.time.DateTime;

import java.util.Locale;

/**
 * kr.kth.commons.timeperiod.TimeFormatter
 * User: sunghyouk.bae@gmail.com
 * Date: 12. 12. 23.
 */
public class TimeFormatter implements ITimeFormatter {

	private static ITimeFormatter instance = new TimeFormatter();
	private static final Object syncLock = new Object();

	// region << Constructors >>


	// endregion

	public static ITimeFormatter getInstance() {
		return instance;
	}

	public static void setInstance(ITimeFormatter formatter) {
		if (formatter == null) return;

		synchronized (syncLock) {
			instance = formatter;
		}
	}

	@Override
	public Locale getLocale() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getListSeparator() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getContextSeparator() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getStartEndSeparator() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getDurationSeparator() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getDateFormat() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getShortDateFormat() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getLongDateFormat() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getShortTimeFormat() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getLongTimeFormat() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public DurationFormatKind getDurationKind() {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public boolean isUseDurationSeconds() {
		return false;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getCollection(int count) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getCollectionPeriod(int count, DateTime start, DateTime end, long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getDate(DateTime date) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getShortDate(DateTime date) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getLongDate(DateTime date) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getTime(DateTime date) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getShortTime(DateTime date) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getLongTime(DateTime date) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getDuration(long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getDuration(long duration, DurationFormatKind durationFormatKind) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getPeriod(DateTime start, DateTime end, long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getInterval(DateTime start, DateTime end, IntervalEdge startEdge, IntervalEdge endEdge, long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getCalendarPeriod(String start, String end, long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getCalendarPeriod(String context, String start, String end, long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}

	@Override
	public String getCalendarPeriod(String startContext, String endContext, String start, String end, long duration) {
		return null;  //To change body of implemented methods use File | Settings | File Templates.
	}
}

File: jvm-experiments-timeperiod/src/main/scala/kr/kth/timeperiod/TimeFormatter.scala
package kr.kth.timeperiod

import IntervalEdge._, DurationFormatKind._
import java.util.Locale
import kr.kth.commons.slf4j.Logging
import org.joda.time.DateTime

/**
 * 시간 정보를 문자열로 포맷팅하는 기본 Time Formatter 입니다.
 * User: sunghyouk.bae@gmail.com
 * Date: 13. 1. 13.
 */
class TimeFormatter extends ITimeFormatter with Logging {

	def getListSeparator: String = ???

	def getContextSeparator: String = ???

	def getStartEndSeparator: String = ???

	def getDurationSeparator: String = ???

	def getDateFormat: String = ???

	def getShortDateFormat: String = ???

	def getLongDateFormat: String = ???

	def getShortTimeFormat: String = ???

	def getLongTimeFormat: String = ???

	def getDurationKind: DurationFormatKind.DurationFormatKind = ???

	def isUseDurationSeconds: Boolean = ???

	def getCollection(count: Int): String = ???

	def getCollectionPeriod(count: Int, start: DateTime, end: DateTime, duration: Long): String = ???

	def getDate(date: DateTime): String = ???

	def getShortDate(date: DateTime): String = ???

	def getLongDate(date: DateTime): String = ???

	def getTime(date: DateTime): String = ???

	def getShortTime(date: DateTime): String = ???

	def getLongTime(date: DateTime): String = ???

	def getDuration(duration: Long, durationFormatKind: DurationFormatKind.DurationFormatKind): String = ???

	def getPeriod(start: DateTime, end: DateTime, duration: Long): String = ???

	def getInterval(start: DateTime, end: DateTime, startEdge: IntervalEdge.IntervalEdge, endEdge: IntervalEdge.IntervalEdge, duration: Long): String = ???

	def getCalendarPeriod(start: String, end: String, duration: Long, startContex: String, endContext: String): String = ???
}

/**
 * 시간 정보를 문자열로 포맷팅하는 Trait 입니다.
 * User: sunghyouk.bae@gmail.com
 * Date: 13. 1. 13.
 */
trait ITimeFormatter {

	def getLocale: Locale = Locale.getDefault

	def getListSeparator: String

	def getContextSeparator: String

	def getStartEndSeparator: String

	def getDurationSeparator: String

	def getDateFormat: String

	def getShortDateFormat: String

	def getLongDateFormat: String

	def getShortTimeFormat: String

	def getLongTimeFormat: String

	def getDurationKind: DurationFormatKind

	def isUseDurationSeconds: Boolean

	def getCollection(count: Int): String

	def getCollectionPeriod(count: Int, start: DateTime, end: DateTime, duration: Long): String

	def getDate(date: DateTime): String

	def getShortDate(date: DateTime): String

	def getLongDate(date: DateTime): String

	def getTime(date: DateTime): String

	def getShortTime(date: DateTime): String

	def getLongTime(date: DateTime): String

	def getDuration(duration: Long, durationFormatKind: DurationFormatKind = DurationFormatKind.Compact): String

	def getPeriod(start: DateTime, end: DateTime, duration: Long): String

	def getInterval(start: DateTime, end: DateTime, startEdge: IntervalEdge, endEdge: IntervalEdge, duration: Long): String

	def getCalendarPeriod(start: String, end: String, startContex: String, endContext: String, duration: Long): String
}

object TimeFormatter extends Logging {

	lazy val instance = new TimeFormatter()
}


